osu.Game/Overlays/BeatmapListingOverlay.cs

```diff
@@ -0,0 +1,288 @@
+                                                        Origin = Anchor.CentreLeft,
+                                                        Margin = new MarginPadding { Left = 20 }
+                                                    }
+                                                }
+                                            },
+                                            panelsPlaceholder = new Container
+                                            {
+                                                AutoSizeAxes = Axes.Y,
+                                                RelativeSizeAxes = Axes.X,
+                                                Padding = new MarginPadding { Horizontal = 20 },
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            var sortCriteria = sortControl.Current;
+            var sortDirection = sortControl.SortDirection;
+
+            searchSection.Query.BindValueChanged(query =>
+            {
+                sortCriteria.Value = string.IsNullOrEmpty(query.NewValue) ? DirectSortCriteria.Ranked : DirectSortCriteria.Relevance;
+                sortDirection.Value = SortDirection.Descending;
+
+                Scheduler.AddOnce(updateSearch);
+            });
+
+            searchSection.Ruleset.BindValueChanged(_ => Scheduler.AddOnce(updateSearch));
+            searchSection.Category.BindValueChanged(_ => Scheduler.AddOnce(updateSearch));
+            sortCriteria.BindValueChanged(_ => Scheduler.AddOnce(updateSearch));
+            sortDirection.BindValueChanged(_ => Scheduler.AddOnce(updateSearch));
+        }
+
+        private void updateSearch()
+        {
+            if (!IsLoaded)
+                return;
+
+            if (State.Value == Visibility.Hidden)
+                return;
+
+            if (API == null)
+                return;
+
+            getSetsRequest?.Cancel();
+            cancellationToken?.Cancel();
+            previewTrackManager.StopAnyPlaying(this);
+
+            panelsPlaceholder.FadeColour(Color4.DimGray, 400, Easing.OutQuint);
+
+            getSetsRequest = new SearchBeatmapSetsRequest(
+                searchSection.Query.Value,
+                searchSection.Ruleset.Value,
+                searchSection.Category.Value,
+                sortControl.Current.Value,
+                sortControl.SortDirection.Value);
+
+            getSetsRequest.Success += response => Schedule(() => recreatePanels(response));
+
+            API.Queue(getSetsRequest);
+        }
+
+        private void recreatePanels(SearchBeatmapSetsResponse response)
+        {
+            var hasError = !string.IsNullOrEmpty(response.Error);
```

Comment:
As far as I can tell from web sources [this is impossible](https://github.com/ppy/osu-web/blob/a02e1cb7dc352f3ee40fecb220d135d726140747/app/Http/Controllers/BeatmapsetsController.php#L106-L111). A non-null `error` in the response will throw a 504 error code, which means it won't raise a `Success` event, but a `Fail` one.

Mind you, arguably the field shouldn't be there at all, but that's for web to handle.